modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -2408,11 +2385,9 @@ private void _updateDescendantNodeObjectDefinitions(
 				String previousRESTContextPath =
 					nodeObjectDefinition.getRESTContextPath();
 
-				nodeObjectDefinition =
-					objectDefinitionLocalService.
-						updateRootDescendantNodeObjectDefinition(
-							nodeObjectDefinition,
-							objectDefinition1.getRootObjectDefinitionId());
+				nodeObjectDefinition.setRootObjectDefinitionIds(
+					objectDefinition1.getRootObjectDefinitionIds(),
+					new long[] {objectRelationship.getObjectDefinitionId2()});
```

Comment:
Can we get the ID from `objectDefinition2` instead of `objectRelationship.getObjectDefinitionId2()`?
I think it is easier to understand.

Reply:
sure